Assets/SEE/Utils/VersionControlSystems.cs

```diff
@@ -0,0 +1,144 @@
+            Blob blob = repo.Lookup<Blob>($"{commitID}:{Path.GetRelativePath(repositoryPath, fileName).Replace("\\", "/")}");
+
+            if (blob != null)
+            {
+                return blob.GetContentText();
+            }
+            else
+            {
+                // File got added and is only available in the other commit.
+                return "";
+            }
+        }
+
+        /// <summary>
+        /// Returns the source code from the revision, that gets compared.
+        /// </summary>
+        /// <returns>the source code from the revision, that gets compared.</returns>
+        public string ShowOriginal(string repositoryPath, string fileName, string oldCommitID, string newCommitID)
+        {
+            Repository repo = new(repositoryPath);
+            Commit newCommit = repo.Lookup<Commit>(newCommitID);
+            Commit oldCommit = repo.Lookup<Commit>(oldCommitID);
+
+            CompareOptions compareOptions = new()
+            {
+                Algorithm = DiffAlgorithm.Myers,
+                Similarity = new SimilarityOptions
+                {
+                    RenameDetectionMode = RenameDetectionMode.Default,
+                }
+            };
+
+            // Compare the commits
+            var changes = repo.Diff.Compare<TreeChanges>(oldCommit.Tree, newCommit.Tree, compareOptions)
+                .Where(change => (change.Path == fileName || change.OldPath == fileName));
+
+            foreach (TreeEntryChanges change in changes)
+            {
+                // Case: file got renamed.
+                if ((change.Status == ChangeKind.Renamed || change.Status == ChangeKind.Copied) &&
+                    (change.Path == fileName || change.OldPath == fileName))
+                {
+                    // Change the filename in the codewindow.
+                    name = change.Path;
+                    // Get renamed file from commit.
+                    var renamedBlob = newCommit[change.Path]?.Target as Blob;
+                    if (renamedBlob != null)
+                    {
+                        return renamedBlob.GetContentText();
+                    }
+                }
+                // Case: File got deleted.
+                if (change.Status == ChangeKind.Deleted && change.Path == fileName)
+                {
+                    // Mark filename in the codewindow as deleted.
+                    name = fileName;
+                    return "";
+                }
+            }
+            Blob blob = newCommit[fileName]?.Target as Blob;
+            return blob.GetContentText();
+        }
+
+        /// <summary>
+        /// Returns the updated filename.
+        /// When a renaming took place, the new name gets returned.
+        /// When a file got deleted, the original filename gets returned.
+        /// Otherwise null gets returned.
+        /// </summary>
+        /// <returns>the updated filename</returns>
+        public string ShowName()
```

Comment:
I am not convinced that having this field `name` is a good idea. `ShowOriginal` has an (unexpected) side effect to set the value of `name`. And only later you query `ShowName`. Meanwhile all sorts of things could have happened. Why cann't `ShowOriginal` return the updated filename?

Reply:
The output of ShowOriginal is the input for the Diff. When we would return the name together with the content of the file, we would need to extract it from the output of the Diff, which may would work, but also overcomplicated I think.
But maybe I am thinking in the wrong direction, could you explain more, how you thought about that?

Reply:
I refactored the API such that this field is no longer needed.